src/query/storages/parquet/src/parquet2/parquet_table/table.rs

```diff
@@ -164,6 +200,125 @@ pub(crate) fn arrow_to_table_schema(mut schema: ArrowSchema) -> TableSchema {
     TableSchema::from(&schema)
 }
 
-pub(super) fn create_parquet_table_info(schema: ArrowSchema) -> TableInfo {
-    naive_parquet_table_info(arrow_to_table_schema(schema).into())
+pub(super) fn create_parquet_table_info(schema: ArrowSchema, stage_info: &StageInfo) -> TableInfo {
+    TableInfo {
+        ident: TableIdent::new(0, 0),
+        desc: "''.'read_parquet'".to_string(),
+        name: format!("read_parquet({})", stage_info.stage_name),
+        meta: TableMeta {
+            schema: arrow_to_table_schema(schema).into(),
+            engine: "SystemReadParquet".to_string(),
+            created_on: Utc.from_utc_datetime(&NaiveDateTime::from_timestamp_opt(0, 0).unwrap()),
+            updated_on: Utc.from_utc_datetime(&NaiveDateTime::from_timestamp_opt(0, 0).unwrap()),
+            ..Default::default()
+        },
+        ..Default::default()
+    }
+}
+
+pub(super) fn blocking_get_parquet2_file_meta(
+    files_to_read: &Option<Vec<StageFileInfo>>,
+    files_info: &StageFilesInfo,
+    operator: &Operator,
+    expect_schema: &SchemaDescriptor,
+    schema_from: &str,
+) -> Result<Vec<FileMetaData>> {
+    let locations = match files_to_read {
+        Some(files) => files
+            .iter()
+            .map(|f| (f.path.clone(), f.size))
+            .collect::<Vec<_>>(),
+        None => files_info
+            .blocking_list(operator, false, None)?
+            .into_iter()
+            .map(|f| (f.path, f.size))
+            .collect::<Vec<_>>(),
+    };
+
+    // Read parquet meta data, sync reading.
+    let mut file_metas = Vec::with_capacity(locations.len());
+    for (location, _) in locations.iter() {
+        let mut reader = operator.blocking().reader(location)?;
+        let file_meta = pread::read_metadata(&mut reader).map_err(|e| {
+            ErrorCode::Internal(format!(
+                "Read parquet file '{}''s meta error: {}",
+                location, e
+            ))
+        })?;
+        check_parquet_schema(expect_schema, file_meta.schema(), location, schema_from)?;
+        file_metas.push(file_meta);
+    }
+    Ok(file_metas)
+}
+
+pub(super) async fn non_blocking_get_parquet2_file_meta(
+    files_to_read: &Option<Vec<StageFileInfo>>,
+    files_info: &StageFilesInfo,
+    operator: &Operator,
+    expect_schema: &SchemaDescriptor,
+    schema_from: &str,
+) -> Result<Vec<FileMetaData>> {
+    let locations = match files_to_read {
+        Some(files) => files
+            .iter()
+            .map(|f| (f.path.clone(), f.size))
+            .collect::<Vec<_>>(),
+        None => files_info
+            .list(operator, false, None)
+            .await?
+            .into_iter()
+            .map(|f| (f.path, f.size))
+            .collect::<Vec<_>>(),
+    };
+
+    // Read parquet meta data, async reading.
+    // TODO(Dousir9): get `max_memory_usage` from ctx.
```

Comment:
It's difficult to get it currently?

Reply:
Yes, this is not important for now, we will solve it later :)